components/mime.rst

```diff
@@ -284,4 +286,317 @@ include them in a message sent with the :doc:`Messenger component
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of dealing with the ``<img src="cid: ..." />`` syntax explained in the
+previous sections, when using Twig to render email contents you can refer to
+image files as usual. First, define a Twig namespace called ``images`` to
+simplify things later::
+
+    // ...
+
+    $templateLoader = new FilesystemLoader(__DIR__.'/templates');
+    $templatedLoader->addPath(__DIR__.'/images', 'images');
+    $twig = new Environment($templateLoader);
+
+Now, use the special ``email.image()`` Twig helper to embed the images inside
+the email contents:
+
+.. code-block:: html+twig
+
+    {# '@images/' refers to the Twig namespace defined earlier #}
+    <img src="{{ email.image('@images/logo.png') }}" />
+
+    <h1>Welcome {{ username }}!</h1>
+    {# ... #}
+
+Attaching Files in Emails with Twig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Similar to embedding images, the first step to attach files using Twig templates
+is to define a dedicated Twig namespace (e.g. ``documents``) to simplify things
+later::
+
+    // ...
+
+    $templateLoader = new FilesystemLoader(__DIR__.'/templates');
+    $templatedLoader->addPath(__DIR__.'/documents', 'documents');
+    $twig = new Environment($templateLoader);
+
+Now, use the special ``email.attach()`` Twig helper to attach files to the email
+message within the Twig template:
+
+.. code-block:: html+twig
+
+    {# '@documents/' refers to the Twig namespace defined earlier #}
+    {% do email.attach('@documents/terms-of-use.pdf') %}
+
+    <h1>Welcome {{ username }}!</h1>
+    {# ... #}
+
+.. note::
+
+    The `Twig do tag`_ is similar to the ``{{ ... }}`` syntax used to evaluate
+    expressions, but it doesn't generate any output. You can use it to set other
+    email properties:
+
+    .. code-block:: html+twig
+
+        {% do email.priority(3) %}
+
+        <h1>Welcome {{ username }}!</h1>
+        {# ... #}
+
+Inlining CSS Styles in Emails with Twig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Designing the HTML contents of an email is very different from designing a
+normal HTML page. For starters, most email clients only support a subset of all
+CSS features. In addition, popular email clients such as Gmail don't support
+defining styles inside ``<style> ... </style>`` sections and you must **inline
+all the CSS styles**.
+
+CSS inlining means that every HTML tag must define a ``style`` attribute with
+all its CSS styles. This not only increases the email byte size significantly
+but also makes it impossible to manage for complex emails. That's why Twig
+provides a ``CssInlinerExtension`` that automates everything for you. First,
+install the Twig extension in your application:
+
+.. code-block:: terminal
+
+    $ composer require twig/cssinliner-extension
+
+Now, enable the extension (this is done automatically in Symfony applications)::
+
+    // ...
+    use Twig\CssInliner\CssInlinerExtension;
+
+    $templateLoader = new FilesystemLoader(__DIR__.'/templates');
```

Comment:
I would simply name it $loader, it’s a bit confusing otherwise, isn’t it? I mean it adds no value to name it this way